cmd/geth/snapshot.go

```diff
@@ -678,10 +678,25 @@ func convertToVerkle(ctx *cli.Context) error {
 		for {
 			select {
 			case <-done:
+				close(flushCh)
 				return
 			default:
-				vRoot.(*verkle.InternalNode).FlushAtDepth(2, saveverkle)
 			}
+
+			v, _ := mem.VirtualMemory()
+			var depth uint8
+			// 1 if > 80%, 2 if 80% < && > 60%...
+			// don't bother cleaning up above 60%
+			depth = 5 - uint8(v.UsedPercent / 20)
+
+			if depth >= 3 {
+				timer := time.NewTimer(time.Minute)
+				<-timer.C	// wait for the tick
+				timer.Stop()
+				continue
+			}
+
```

Comment:
As long as you're not hitting the if-clause above, you're effectively spinning on `vRoot.(*verkle.InternalNode).FlushAtDepth` and `mem.VirtualMemory()`. 

That can't be good, the latter might be a `syscall` too?

Reply:
if the `if` clause isn't hit, it means there's a lot of data to flush, so the `syscall` (which it most definitely is, it's calling `/proc` on Linux and `sysctl` on *BSD) is called at most once per minute, iff the memory usage is below 60%.

I'm working on your suggestions though, just saved the current state of what was running on my test machines, in case we need to get back to it.